shipyard_blueprints/box/shipyard_box/cli/download.py

```diff
@@ -207,41 +194,53 @@ def get_client(service_account):
         client.user().get()
         return client
     except BoxOAuthException as e:
-        print(
-            f"Error accessing Box account with pervice account "
-            f"developer_token={developer_token}; client_id={client_id}; "
-            f"client_secret={client_secret}"
+        print("Error accessing Box account with service account ")
+        raise e
+
+
+def get_folder(client, folder_name):
+    """
+    Returns the folder id for the Box client if it exists.
+    """
+    try:
+        print(f"Attempting to find folder {folder_name}")
+        folders = client.search().query(query=folder_name, type="folder")
+        base_folder = os.path.basename(os.path.normpath(folder_name))
+
+        return next(
+            (folder for folder in folders if folder.name == base_folder),
+            None,
         )
-        raise (e)
+    except (BoxOAuthException, BoxAPIException) as e:
+        raise e
 
 
-def get_file_id(client, source_folder_name, source_file_name):
+def get_file_id(client, source_file_name, source_folder_name=None):
     """
     Returns the file id for the Box client if it exists.
     """
     try:
-        search_folder = None
         if not source_folder_name:
-            search_folder = client.folder("0")
+            folder_id = "0"
         else:
-            folders = client.search().query(query=source_folder_name, type="folder")
-            for folder in folders:
-                search_folder = folder
-
-        files = client.search().query(
-            query=source_file_name, type="file", ancestor_folders=[search_folder]
-        )
+            folder = get_folder(client, source_folder_name)
+            folder_id = folder.id
+            print(f"Found folder {source_folder_name} with ID {folder_id}")
+        if not folder_id:
+            raise FileNotFoundError(f"Folder {source_folder_name} not found")
+        files = client.folder(folder_id).get_items()
         for _file in files:
-            return (_file.name, _file.id)
+            if _file.name == source_file_name:
+                print(f"Found file {source_file_name} with ID {_file.id}")
+                return _file.name, _file.id
+
+        raise FileNotFoundError(f"File {source_file_name} not found in folder {source_folder_name}")
```

Comment:
Part of the issue that customers were seeing was than if the folder was found but no files existed within the folder it returned a difficult to read error. I included logging to assist with the troubleshooting process and added logs in plain english